Extensions/Video/videoruntimeobject.js

```diff
@@ -0,0 +1,294 @@
+  return val <= min ? min : val >= max ? max : val;
+};
+
+/**
+ * Set the volume of the video object.
+ * @param {number} volume The new volume.
+ */
+gdjs.VideoRuntimeObject.prototype.setVolume = function(volume) {
+  this._volume = this._clamp(this._normalize(volume, 0, 100), 0, 1) * 100;
+  this._renderer.updateVolume();
+};
+
+/**
+ * Get the volume of the video object.
+ */
+gdjs.VideoRuntimeObject.prototype.getVolume = function() {
+  return this._normalize(this._renderer.getVolume(), 0, 1) * 100;
+};
+
+/**
+ * Check if the video is being played.
+ */
+gdjs.VideoRuntimeObject.prototype.isPlayed = function() {
+  return this._renderer.isPlayed();
+};
+
+/**
+ * Check if the video is paused.
+ */
+gdjs.VideoRuntimeObject.prototype.isPaused = function() {
+  return !this._renderer.isPlayed();
+};
+
+/**
+ * Check if the video is looping.
+ */
+gdjs.VideoRuntimeObject.prototype.isLooped = function() {
+  return this._renderer.isLooped();
+};
+
+/**
+ * Return the total time of the video.
+ */
+gdjs.VideoRuntimeObject.prototype.getDuration = function() {
+  return this._renderer.getDuration();
+};
+
+/**
+ * Check if the video has ended.
+ */
+gdjs.VideoRuntimeObject.prototype.isEnded = function() {
+  return !this._renderer.isEnded();
+};
+
+/**
+ * Set the new time of the video object.
+ * @param {number} time The new time.
+ */
+gdjs.VideoRuntimeObject.prototype.setCurrentTime = function(time) {
+  this._renderer.setCurrentTime(time);
+};
+
+/**
+ * Get the current time of the video object.
+ */
+gdjs.VideoRuntimeObject.prototype.getCurrentTime = function() {
+  return this._renderer.getCurrentTime();
+};
+
+/**
+ * Set the new playback speed of the video object.
+ * @param {number} playbackSpeed The new playback speed.
+ */
+gdjs.VideoRuntimeObject.prototype.setPlaybackSpeed = function(playbackSpeed) {
+  this._playbackSpeed = this._clamp(playbackSpeed, 0.5, 2);
+  this._renderer.setPlaybackSpeed(this._playbackSpeed);
+};
+
+/**
+ * Get the playback speed of the video object.
+ */
+gdjs.VideoRuntimeObject.prototype.getPlaybackSpeed = function() {
+  return this._renderer.getPlaybackSpeed();
+};
+
+gdjs.RuntimeScene.gdjsCallbackRuntimeScenePaused = function(runtimeScene) {
+  // TODO: Add method to get list of all instances
+  for (var instances in runtimeScene._instances.items) {
+    for (var object in runtimeScene._instances.items[instances]) {
+      var obj = runtimeScene._instances.items[instances][object];
+      // TODO: Move this to renderer
+      // TODO: Use instanceof instead of type
+      if (
+        obj.type == "Video::VideoObject" &&
+        !obj._renderer._pixiObject._texture.baseTexture.source.paused &&
+        !obj._renderer._pixiObject._texture.baseTexture.source.ended
+      ) {
+        obj._renderer._pixiObject._texture.baseTexture.source.pause();
+        obj._pausedAsScenePaused = true;
+      }
+    }
+  }
+};
+
+gdjs.RuntimeScene.gdjsCallbackRuntimeSceneResumed = function(runtimeScene) {
```

Comment:
These methods need to be on gdjs.VideoRuntimeObject, not on gdjs.RuntimeScene - otherwise, any other extension adding the same callbacks will be conflicting with these ;)